src/main/java/io/takari/maven/builder/smart/ReactorBuildQueue.java

```diff
@@ -43,8 +43,8 @@ public ReactorBuildQueue(Collection<MavenProject> projects,
       }
     }
 
-    this.rootProjects = ImmutableSet.copyOf(rootProjects);
-    this.projects = ImmutableSet.copyOf(projects);
+    this.rootProjects = unmodifiableSet(rootProjects);
+    this.projects = unmodifiableSet(new HashSet<>(projects));
```

Comment:
here you DO create a copy

Reply:
Yes, because `projects` is a `Collection<>`, not a `List` or `Set`.